Allow to close opened diagram tabs

SteveMonnier opened the pull request:
A blue cross icon has been added on the diagram tab to be able to close
them.
RAF: Closing the last opened diagram automatically reopens it because of
the representationId kept by the method useParams in EditProjectView.
Also, there is some extra space after the cross icon that should be
removed.

Change-Id: I4d2b8532b4df312c4526cc91d1a9fe525f3d6860
Signed-off-by: Steve Monnier <steve.monnier@obeo.fr>

sbegaudeau commented on frontend/src/views/edit-project/EditProjectView.tsx:
```diff
@@ -109,6 +110,11 @@ export const EditProjectView = () => {
     if (displayedRepresentation && displayedRepresentation.id !== representationId) {
       const pathname = generatePath(routeMatch.path, { projectId, representationId: displayedRepresentation.id });
       history.push({ pathname });
+    } else if (displayedRepresentation === undefined) {
```
`if (!displayedRepresentation) {` in order to both consider `null` and `undefined`

SteveMonnier replied:
Done

sbegaudeau commented on frontend/src/views/edit-project/RepresentationNavigation.module.css:
```diff
@@ -17,6 +17,7 @@
 }
 
 .item {
+  columns: 2;
```
Do not use CSS `column` layout, start with Flexbox and if really necessary to Grid

SteveMonnier replied:
Done

sbegaudeau commented on frontend/src/views/edit-project/__tests__/reducer.test.ts:
```diff
@@ -295,6 +309,38 @@ describe('EditProjectView - reducer', () => {
+      message: undefined,
+    });
+  });
+
```
Could you add a test for the last possibility of your `if / else` in `handleRepresentationClosedAction`?

SteveMonnier replied:
Done

sbegaudeau reviewed (changes requested):
- You do not need to use a changeId, we are not using Gerrit anymore
- Please reference a Github issue in your commit message

sbegaudeau reviewed (changes requested):
Sorry, wrong button -.-

SteveMonnier commented:
> * o

SteveMonnier commented:
I am not sure how to reply to one of your previous comment directly, but I removed the  changeId I added by reflex and added the reference to a Github issue I just created.

sbegaudeau reviewed (changes requested):
The second commit regarding edge edition does not belong in this pull request

pcdavid commented on frontend/src/views/edit-project/EditProjectView.tsx:
```diff
@@ -109,6 +110,12 @@ export const EditProjectView = () => {
     if (displayedRepresentation && displayedRepresentation.id !== representationId) {
       const pathname = generatePath(routeMatch.path, { projectId, representationId: displayedRepresentation.id });
       history.push({ pathname });
+    } else if (!displayedRepresentation && performance.navigation.type !== PerformanceNavigation.TYPE_RELOAD) {
```
We should not have to depend on the "performance" API here. I guess the issue (and fix) is similar to what you did for the OnboardArea bug, i.e. refine the state machine.

pcdavid commented on frontend/src/views/edit-project/EditProjectView.tsx:
```diff
@@ -109,6 +110,12 @@ export const EditProjectView = () => {
       const pathname = generatePath(routeMatch.path, { projectId, representationId: displayedRepresentation.id });
       history.push({ pathname });
+    } else if (!displayedRepresentation && performance.navigation.type !== PerformanceNavigation.TYPE_RELOAD) {
+      //If the browser was reloaded there is no need to generate a new path
```
The issue is not about reloading.

sbegaudeau commented:
Will be merged as part of https://github.com/eclipse-sirius/sirius-components/pull/294
